ingest/perf_test_ingestor.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2019 Rackspace US, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ingest
+
+import (
+	"context"
+	"fmt"
+	"github.com/racker/salus-telemetry-protocol/telemetry_edge"
+	"github.com/racker/telemetry-envoy/ambassador"
+	"github.com/racker/telemetry-envoy/config"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type PerfTestIngestor struct {
+	egressConn       ambassador.EgressConnection
+	metricsPerMinute int
+	floatsPerMetric  int
+	ticker           *time.Ticker
+	newRateC         chan int
+}
+
+func init() {
+	registerIngestor(&PerfTestIngestor{})
+}
+
+func (p *PerfTestIngestor) Bind(conn ambassador.EgressConnection) error {
+	if viper.GetInt(config.PerfTestPort) == 0 {
+		return nil
+	}
+	log.Info("entering perfTest mode")
+	p.egressConn = conn
+	p.metricsPerMinute = 60
+	p.floatsPerMetric = 10
+	p.newRateC = make(chan int)
+	return nil
+}
+
+func (p *PerfTestIngestor) Start(ctx context.Context) {
+	if viper.GetInt(config.PerfTestPort) == 0 {
+		return
+	}
+
+	p.ticker = time.NewTicker(time.Duration(int64(time.Minute) / int64(p.metricsPerMinute)))
+	go p.startPerfTestServer()
+	for {
+		select {
+		case <-ctx.Done():
+			p.ticker.Stop()
+			return
+		case p.metricsPerMinute = <-p.newRateC:
+			p.ticker.Stop()
+			p.ticker = time.NewTicker(time.Duration(int64(time.Minute) / int64(p.metricsPerMinute)))
+		case <-p.ticker.C:
+			p.processMetric()
+		}
+	}
+}
+
+func (p *PerfTestIngestor) startPerfTestServer() {
+	serverMux := http.NewServeMux()
+	serverMux.Handle("/", http.HandlerFunc(p.handler))
+
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", viper.GetInt(config.PerfTestPort)))
+	if err != nil {
+		log.Fatalf("couldn't create perf test server")
+	}
+	log.Info("started perfTest webServer")
+	err = http.Serve(listener, serverMux)
+	log.Fatalf("perf test server error %v", err)
+}
+
+func (p *PerfTestIngestor) handler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
There's actually a status code for that
```suggestion
		w.WriteHeader(http.StatusMethodNotAllowed)
```